### PR TITLE
Shortcuts by @te-online

### PR DESCRIPTION
**Pull request by mario**
@te-online I've had trouble properly rebasing your branch, so I've done it like this.

I've improved a bit on your awesome work. Would you be so kind to review both the code and the functionality here?

Please note that the shortcuts are created based on CURRENT USER and LAST ACTIVITY. I completely ignored favorites conversations. Unsure if that's wise, so I'm open to opinions.

I'd really really appreciate it.

### CODE REVIEW

**Comment by mario** [comment]
@ma12-co your opinion is welcome here as well. Currently we show last 3 active conversations (last activity) in Shortcuts. Do you think I should take "favorite" value into consideration here as well?

**Comment by ma12-co** [comment]
> @ma12-co your opinion is welcome here as well. Currently we show last 3 active conversations (last activity) in Shortcuts. Do you think I should take "favorite" value into consideration here as well?

I think that favourite values should have priority over the latest convs :)

**Comment by mario** [comment]
Consider it done @ma12-co!

**Comment by ma12-co** [comment]
This android-bot is a bit intrusive :sweat_smile:

**Comment by mario** [comment]
@ma12-co "bit"? :-/

**Comment by te-online** [comment]
@mario Thanks so much for your work and kind words 😊 
I'll check it out asap, just to see how you did it. And I agree with @ma12-co that it could be nice to align the list of chats in the app and the shortcuts so that a users mental modal of the list stays the same.

**Comment by mario** [comment]
@te-online looking forward to your generous feedback! :)

**Comment by mario** [comment]
I'll  merge this to make it easier to continue development, but after your feedback @te-online we can continue fixing and improving on this.

Once again thank yo for your awesome contribution, this wouldn't have happened without you.

File: app/src/main/java/com/nextcloud/talk/activities/MainActivity.kt [hunk 1]
```diff
@@ -152,7 +127,24 @@ class MainActivity : BaseActivity(), ActionBarProvider {
     override fun onNewIntent(intent: Intent) {
         super.onNewIntent(intent)
 
-        if (intent.hasExtra(BundleKeys.KEY_FROM_NOTIFICATION_START_CALL)) {
+        if (intent.action == BundleKeys.KEY_NEW_CONVERSATION) {
+            openNewConversationScreen()
+        } else if (intent.action == BundleKeys.KEY_OPEN_CONVERSATION) {
+            GlobalScope.launch {
+                val user: UserNgEntity? = usersRepository.getUserWithId(intent.getLongExtra(BundleKeys.KEY_INTERNAL_USER_ID, -1))
+                user?.let {
```
Comment (te-online) [inline app/src/main/java/com/nextcloud/talk/activities/MainActivity.kt]: I'd been running into the restrictions of `persistable bundle` and was thinking to only put the user id into the intent, but I'd have never figured this out, thanks! 😂

File: app/src/main/java/com/nextcloud/talk/newarch/utils/ShortcutService.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        openNewConversationIntent.action = BundleKeys.KEY_NEW_CONVERSATION
+
+        val shortcuts: MutableList<ShortcutInfo> = mutableListOf()
+        val images = Images()
+
+        currentUser?.let { user ->
+            shortcuts.add(ShortcutInfo.Builder(context, "new")
+                    .setShortLabel(context.resources.getString(R.string.nc_new_conversation_short))
+                    .setLongLabel(context.resources.getString(R.string.nc_new_conversation))
+                    .setIcon(Icon.createWithBitmap(context.resources.getDrawable(R.drawable.new_conversation_shortcut).toBitmap()))
```
Comment (te-online) [inline app/src/main/java/com/nextcloud/talk/newarch/utils/ShortcutService.kt]: A so-called _one-liner_ 🙈 But the result is nice!

**Comment by mario** [comment]
@te-online let me know what you'd like to work on next, more than happy to assist you. :)

**Comment by te-online** [comment]
Glad I was able to help. Thanks for fixing the mess and making it work. 🎉  Your changes give me a good sense of what I was missing 😊 

I'll try to look at one of the open issues – hopefully soon 👍

**Comment by mario** [comment]
You are awesome - your code was actually quite good considering lack of knowledge about the overall app and the mess it's currently in with all the refactoring going on.

Looking forward to working more with you and I hope to see you at one of our contributor weeks!

File: app/src/main/java/com/nextcloud/talk/activities/MainActivity.kt (hunk 1 above)
Comment (mario) [reply app/src/main/java/com/nextcloud/talk/activities/MainActivity.kt]: :) I'd still argue we shouldn't be putting the whole user entity into the bundle, but for now it'll do as I do this all across the codebase anyway xD

**Pull request closed without merging**